source/common/http/http1/codec_impl.cc

```diff
@@ -407,34 +413,54 @@ void RequestEncoderImpl::encodeHeaders(const RequestHeaderMap& headers, bool end
   encodeHeadersBase(headers, absl::nullopt, end_stream);
 }
 
+int ConnectionImpl::setAndCheckCallbackStatus(Status&& status) {
+  codec_status_ = std::move(status);
+  return codec_status_.ok() ? enumToInt(HttpParserCode::Success) : enumToInt(HttpParserCode::Error);
+}
+
+int ConnectionImpl::setAndCheckCallbackStatusOr(Envoy::StatusOr<int>&& statusor) {
+  if (statusor.ok()) {
```

Comment:
Ditto about ASSERT(codec_status_.ok()) at the top of the method.

Reply:
done